emily/handler/src/api/handlers/new_block.rs

```diff
@@ -0,0 +1,580 @@
+/// populated using the deployer variable in the config.
+///
+/// Although the stacks node is supposed to only send sbtc-registry events,
+/// the node can be misconfigured or have some bug where it sends other
+/// events as well. Accepting such events would be a security issue, so we
+/// filter out events that are not from the sbtc-registry.
+///
+/// See https://github.com/stacks-network/sbtc/issues/501.
+static SBTC_REGISTRY_IDENTIFIER: OnceLock<QualifiedContractIdentifier> = OnceLock::new();
+
+/// Maximum request body size for the event observer endpoint.
+///
+/// Stacks blocks have a limit of 2 MB, which is enforced at the p2p level, but
+/// event observer events can be larger than that since they contain the
+/// subscribed sbtc events. Luckily, the size of the sbtc events themselves are
+/// bounded by the size of the transactions that create them, so a limit of 8 MB
+/// will be fine since it is twice as high as required.
+pub const EVENT_OBSERVER_BODY_LIMIT: usize = 8 * 1024 * 1024;
+
+#[derive(Clone)]
+struct StacksBlock {
+    pub block_hash: String,
+    pub block_height: u64,
+}
+
+/// Get limits handler.
+#[utoipa::path(
+    post,
+    operation_id = "newBlock",
+    path = "/new_block",
+    tag = "new_block",
+    request_body = NewBlockEventRaw,
+    responses(
+        (status = 200, description = "New Block event received successfully"),
+        (status = 400, description = "Invalid request body", body = ErrorResponse),
+        (status = 404, description = "Address not found", body = ErrorResponse),
+        (status = 405, description = "Method not allowed", body = ErrorResponse),
+        (status = 500, description = "Internal server error", body = ErrorResponse)
+    ),
+    security(("ApiGatewayKey" = []))
+)]
+#[instrument(skip_all, name = "new-block")]
+pub async fn new_block(
+    context: EmilyContext,
+    new_block_event: NewBlockEventRaw,
+) -> impl warp::reply::Reply {
+    // Internal handler so `?` can be used correctly while still returning a reply.
+    async fn handler(
+        context: EmilyContext,
+        new_block_event: NewBlockEventRaw,
+    ) -> Result<impl warp::reply::Reply, Error> {
+        let new_block_event = match new_block_event.deserialize() {
+            Ok(event) => event,
+            Err(error) => {
+                tracing::error!(%error, "failed to deserialize new block event");
+                return Err(error.into());
+            }
+        };
+        // Set the global limits.
+        let registry_address = SBTC_REGISTRY_IDENTIFIER.get_or_init(|| {
+            // Although the following line can panic, our unit tests hit this
+            // code path so if tests pass then this will work in production.
+            let contract_name = ContractName::from(SBTC_REGISTRY_CONTRACT_NAME);
+            let issuer = StandardPrincipalData::from(context.settings.deployer_address);
+            QualifiedContractIdentifier::new(issuer, contract_name)
+        });
```

Comment:
I don't think that we need to remove this, but I'm not sure that this buys us anything in production since it gets executed in a lambda.